controllers/util/k8sutil.go

```diff
@@ -148,3 +153,19 @@ func GenerateNameForNonResilientCephBlockPoolSC(initData *ocsv1.StorageCluster)
 	}
 	return fmt.Sprintf("%s-ceph-non-resilient-rbd", initData.Name)
 }
+
+func GetCSVWithPrefix(ctx context.Context, kubeClient client.Client, logger *logr.Logger, prefix string, namespace string) (*opv1a1.ClusterServiceVersion, error) {
+	csvList := &opv1a1.ClusterServiceVersionList{}
+	err := kubeClient.List(ctx, csvList, client.InNamespace(namespace))
+	if err != nil {
+		logger.Error(err, "failed to list csvs", "namespace", namespace)
```

Comment:
Having a logger as a requirement for utils is bad practice. The alternative is to wrap the error with some context (which is always a good practice), return it, then log the error on the call site.

Reply:
removed logger from utils, returning the formatted error and logging at call site